### ISSUE TITLE
Where is the semi supervision in train

### ISSUE THREAD

**Issue by lanlandefan** [zuoym15/craves.ai#18]
Hello!Your work is wonderful and gives me a lot of inspiration. 
     But I have a question:where is the semi supervision algorithm in train？The paper says that you incorporate iterative algorithm with stochastic gradient descent.
    Thank you very much!
![5f94c27a2b61668143c0225703254c8](https://user-images.githubusercontent.com/37535683/144359556-752a54fa-e3bb-4e14-a10b-71bab093890e.png)
